fontforge/parsettf.c

```diff
@@ -4720,13 +4720,16 @@ return;
 	    enc = FindOrMakeEncoding("UnicodeFull");
 
 	if ( format==0 ) {
+	    if (len > 256 + 6) {
+		IError("Table too large; truncated to 256 entries.");
+	    }
 	    if ( justinuse==git_normal && map!=NULL && map->enccount<256 ) {
 		map->map = realloc(map->map,256*sizeof(int));
-		memset(map->map,-1,(256-map->enccount)*sizeof(int));
+		memset(map->map+map->enccount,-1,(256-map->enccount)*sizeof(int));
 		map->enccount = map->encmax = 256;
 	    }
 	    for ( i=0; i<len-6; ++i )
-		table[i] = getc(ttf);
+		if (i < 256) table[i] = getc(ttf); else getc(ttf);
```

Comment:
Since there's nothing else in this subtable except the mappings, I don't think there's a need to call `getc(ttf)` on the `else` condition (and it's questionable whether or not you'd want to read out the whole amount anyway if you suspect that the length field is corrupt).

I'd say it'd be sufficient to just set `len=256 + 6` if it exceeds that value.

Reply:
I did not check the call path, but there might be one that follows this read with a non-seeking read. If not now, maybe in the future. It seems appropriate to burn the rest of the bytes corresponding to this subtable before returning control.

Reply:
Hmm I'd be very surprised if the next function to read off didn't do a seeking read

Reply:
Perhaps. But it's not incorrect, and it's not worth changing at this point. I still think it's a good safety feature for the future.

Reply:
Hmm, ok, but that's definitely not a safety feature; at best it's just one approach to undefined behaviour. Since it can already be inferred that the length is invalid, reading past the end of the max permissible length may in fact be overrunning whatever data follows this subtable.

In either case, without a seeking read, it's not going to make much sense to read whatever's there, so it kind of seems pointless to try to read past what you know should be the max length.